packages/node-cli/src/commands/ManageAccountCommandGenerator.js

```diff
@@ -0,0 +1,288 @@
+               value: ACTION.EXIT,
+            },
+         ],
+      });
+
+      if (answer[ANSWERS_NAMES.ACTION] == ACTION.EXIT) {
+         throw NodeTranslationService.getMessage(MESSAGES.CANCEL);
+      }
+
+      return answer[ANSWERS_NAMES.ACTION];
+   }
+
+   async _introduceNewName(prompt, authIDList, originalAuthId) {
+      const answer = await prompt({
+         type: CommandUtils.INQUIRER_TYPES.INPUT,
+         name: ANSWERS_NAMES.RENAMETO,
+         message: NodeTranslationService.getMessage(QUESTIONS.NEW_NAME),
+         filter: (fieldValue) => fieldValue.trim(),
+         validate: (fieldValue) =>
+            showValidationResults(
+               fieldValue,
+               validateFieldIsNotEmpty,
+               validateFieldHasNoSpaces,
+               (fieldValue) => validateSameAuthID(fieldValue, originalAuthId),
+               (fieldValue) => validateAuthIDNotInList(fieldValue, Object.keys(authIDList)),
+               validateAlphanumericHyphenUnderscore,
+               validateMaximumLength
+            ),
+      });
+      return answer[ANSWERS_NAMES.RENAMETO];
+   }
+
+   async _confirmRemove(prompt) {
+      const answer = await prompt([
+         {
+            type: CommandUtils.INQUIRER_TYPES.LIST,
+            name: ANSWERS_NAMES.REMOVE,
+            message: NodeTranslationService.getMessage(QUESTIONS.VERIFY_REMOVE),
+            default: false,
+            choices: [
+               { name: NodeTranslationService.getMessage(YES), value: true },
+               { name: NodeTranslationService.getMessage(NO), value: false },
+            ],
+         },
+      ]);
+      if (!answer[ANSWERS_NAMES.REMOVE]) {
+         throw NodeTranslationService.getMessage(MESSAGES.CANCEL);
+      }
+      return answer[ANSWERS_NAMES.REMOVE];
+   }
+
+   _extractAnswers(answers) {
+      if (answers[ANSWERS_NAMES.ACTION] == ACTION.RENAME) {
+         return {
+            [COMMAND.OPTIONS.RENAME]: answers[ANSWERS_NAMES.SELECTED_AUTH_ID].authId,
+            [COMMAND.OPTIONS.RENAMETO]: answers[ANSWERS_NAMES.RENAMETO],
+         };
+      } else if (answers[ANSWERS_NAMES.ACTION] == ACTION.REMOVE) {
+         return {
+            [COMMAND.OPTIONS.REMOVE]: answers[ANSWERS_NAMES.SELECTED_AUTH_ID].authId,
+         };
+         // } else if (answers[ANSWERS_NAMES.ACTION] == ACTION.REVOKE) {
+         //    return {
+         //    [COMMAND.OPTIONS.REVOKE]: answers[ANSWERS_NAMES.SELECTED_AUTH_ID].authId,
+         // }
+      }
+   }
+
+   async _executeAction(answers) {
+      const sdkParams = CommandUtils.extractCommandOptions(answers, this._commandMetadata);
+      const flags = [];
+      if (answers[COMMAND.OPTIONS.LIST]) {
+         flags.push(COMMAND.OPTIONS.LIST);
+         delete sdkParams[COMMAND.OPTIONS.LIST];
+      }
+      const executionContext = new SdkExecutionContext({
+         command: this._commandMetadata.sdkCommand,
+         params: sdkParams,
+         flags,
+      });
+
```

Comment:
we could have:
`const actionExecuted = this.getActionExecuted(answers)`

and then use this action when calling `this._getSpinnerMessage(actionExecuted)`
So that we could simplify `_getSpinnerMessage` function to:
```
_getSpinnerMessage(action) {
      switch(action) {
            case ACTION.REMOVE:
                  return NodeTranslationService.getMessage(MESSAGES.REMOVING);
                  break;
            case ACTION.RENAME:
                  return NodeTranslationService.getMessage(MESSAGES.RENAMING);
                  break;
            case ACTION.LIST:
                  return NodeTranslationService.getMessage(MESSAGES.LISTING);
                  break
            case ACTION.REVOKE:
                  return NodeTranslationService.getMessage(MESSAGES.REVOKING);
                  break;
            default:
                  return NodeTranslationService.getMessage(MESSAGES.INFO, answers.info);
      }
   }
```

and we could also modify the cal to:
`this.prepareData(actionExecuted, operationResult.data)` 
and 
`.withActionExecuted(actionExecuted)`